### PR TITLE
Update dependancies and release 0.19

### LINKED ISSUE #112: Conflicting implementations on master
Currently on master I get:
```
error[E0119]: conflicting implementations of trait `std::hash::Hash` for type `zkp::Tag`
   --> /home/casatta/.cargo/registry/src/github.com-1ecc6299db9ec823/secp256k1-zkp-sys-0.4.0/src/zkp.rs:422:1
    |
422 | impl_array_newtype!(Tag, c_uchar, 32);
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ conflicting implementation for `zkp::Tag`
...
437 | impl hash::Hash for Tag {
    | ----------------------- first implementation here
    |
    = note: this error originates in the macro `impl_array_newtype` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0119]: conflicting implementations of trait `std::hash::Hash` for type `zkp::PedersenCommitment`
   --> /home/casatta/.cargo/registry/src/github.com-1ecc6299db9ec823/secp256k1-zkp-sys-0.4.0/src/zkp.rs:458:1
    |
458 | impl_array_newtype!(PedersenCommitment, c_uchar, 64);
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ conflicting implementation for `zkp::PedersenCommitment`
...
473 | impl hash::Hash for PedersenCommitment {
    | -------------------------------------- first implementation here
    |
    = note: this error originates in the macro `impl_array_newtype` (in Nightly builds, run with -Z macro-backtrace for more info)

For more information about this error, try `rustc --explain E0119`.
error: could not compile `secp256k1-zkp-sys` due to 2 previous errors
warning: build failed, waiting for other jobs to finish...
error: build failed

```

a temporary workaround is to pin secp256k1-sys in Cargo.toml
```
secp256k1-sys = "=0.4.1"
```

I am not sure what the right fix is but probably it needs a release of secp256k1-sys ?

### PR DESCRIPTION
**Pull request by sanket1729**
Dependency upgrade and taproot support implemented in #121. Please review that before making a release. 

This PR is now only has a single commit with the release. 

Fixes #112

### CODE REVIEW

**Comment by sanket1729** [comment]
Please look at #118 before merging this.

**Review by apoelstra — APPROVED** [review]
ACK 24c8b78f94c52c59f34d8a63d3f52672cfe1ffb9

@RCasatta ok to merge this? anything else we should get into this major release?

**Comment by sanket1729** [comment]
@apoelstra, While updating elements-miniscript. I realized that I also need psbt support here. Converting to draft to avoid merging this.

**Comment by RCasatta** [comment]
> Converting to draft to avoid merging this.

Maybe this will buy us some time to release bitcoind and elementsd with bitcoin 0.28 once rust-bitcoincore-rpc is published (so that we can remove github ref)

File: Cargo.toml
Comment (RCasatta) [inline Cargo.toml:40]: elementsd `0.5.0`  has been released with updated bitcoins with bitcoin `0.28.0`

Comment (sanket1729) [reply Cargo.toml:40]: Thanks. Will update the PR

**Comment by sanket1729** [comment]
Will rebase only after #121 is merged.

**Comment by sanket1729** [comment]
This final release PR is ready for review. @apoelstra

**Comment by sanket1729** [comment]
Converted to draft until I complete the elements-miniscript rebase. I realize that I might need many more things from here.

**Comment by sanket1729** [comment]
Please review #123 before this one.

**Comment by apoelstra** [comment]
Ok, will ack #123 once my tests finish running ... but I think we will need a followup after that to address Steven's nits?

**Comment by sanket1729** [comment]
Marked ready for review. @apoelstra, this can be merged after #125 which addresses Steven's nits

**Review by apoelstra — APPROVED** [review]
ACK 948254b18235cf27d47553c86762413a741a42f3

**Comment by apoelstra** [comment]
Tagged and published.

**Pull request merged**
